### ISSUE TITLE
Sum as an aggregation function

### ISSUE THREAD

**Issue by pegger0709** [jpmml/sklearn2pmml#192]
Hello,
I am trying to create a pipeline where certain groups of columns are aggregated by averaging them, while other columns are aggregated by summing them. While the latter is doable, the former is not, as it throws an IllegalArgumentException.
To be more specific, I have a pandas data frame with some input data from patients, and I want to classify them to the appropriate treatment. 
Certain columns pertain to different exercises designed to assess motor function, I want to sum these columns up to one "MOTOR_FUNCTION' column. Other columns pertain to different self-assessments of spasticity, and I want to average these columns into one "SPASTICITY" column. Also, there is one column indicating whether or not the patient is open to experimental treatments, I want to leave that column alone.
Finally, I want to train a decision tree classifier on this reduced patient input, and write the pipeline to PMML so the production team can test it.
Here is an idea of the code I used

```
mapper = DataFrameMapper([
    (MOTOR_FUNCTION, Aggregator(function='sum'), {'alias' : 'MOTOR_FUNCTION'}),
    (SPASTICITY, Aggregator(function='mean'), {'alias' : 'SPASTICITY'}),
    ('EXPERIMENTAL', Aggregator(function=None))
])
clf = DecisionTreeClassifier()

pipeline = PMMLPipeline([
    ("mapper", mapper),
    ("classifier", clf)
])

pipeline.fit(patient_input, y)
sklearn2pmml(pipeline, "../Practice/PMML/DecisionTreeStS.pmml", with_repr = True)
```

The output is as follows:

```
SEVERE: Failed to convert
java.lang.IllegalArgumentException: sum
	at sklearn2pmml.preprocessing.Aggregator.translateFunction(Aggregator.java:84)
	at sklearn2pmml.preprocessing.Aggregator.encodeFeatures(Aggregator.java:56)
	at sklearn.Transformer.updateAndEncodeFeatures(Transformer.java:85)
	at sklearn_pandas.DataFrameMapper.initializeFeatures(DataFrameMapper.java:73)
	at sklearn.Initializer.encodeFeatures(Initializer.java:41)
	at sklearn.Transformer.updateAndEncodeFeatures(Transformer.java:85)
	at sklearn.pipeline.Pipeline.encodeFeatures(Pipeline.java:83)
	at sklearn2pmml.pipeline.PMMLPipeline.encodePMML(PMMLPipeline.java:203)
	at org.jpmml.sklearn.Main.run(Main.java:145)
	at org.jpmml.sklearn.Main.main(Main.java:94)
```

Is there an a priori reason why aggregating by averaging is OK but aggregating by summing is not? It would be great to be able to do both.
Thanks so much.
Philip

**Comment by vruusmann** [comment]
> there is one column indicating whether or not the patient is open to experimental treatments, I want to leave that column alone.

Right now you're using the following construct for this:
``` python
DataFrameMapper([
    ('EXPERIMENTAL', Aggregator(function=None))
])
```

The `sklearn_pandas.DataFrameMapper` transformer supports `None` as a transformer. By performing this replacement, the above simplifies to:
```python
DataFrameMapper([
    ('EXPERIMENTAL', None)
])
```

> Is there an a priori reason why aggregating by averaging is OK but aggregating by summing is not?

As for this "main question", then the underlying JPMML-SkLearn library throws `java.lang.IllegalArgumentException: sum` to indicate that the "sum" aggregation method has not been implemented into the `sklearn2pmml.preprocessing.Aggregator` transformer class yet:
https://github.com/jpmml/sklearn2pmml/blob/master/sklearn2pmml/preprocessing/__init__.py#L21-L39

This wouldn't be much work, and I'll try to get it into the next JPMML-SkLearn/SkLearn2PMML release (should happen sometimes this week already).

In the meantime, you can "emulate" summation using `sklearn2pmml.preprocessing.ExpressionTransformer`. Something like that:
```python
mapper = DataFrameMapper([
    (MOTOR_FUNCTION, ExpressionTransformer("X[0] + X[1] + X[2]")
])
```

Not very elegant (you'd need to hard-code the length of motor length columns), but at least you can keep exploring the other parts of your workflow.

**Comment by pegger0709** [comment]
That's wonderful, thank you very much!

**Comment by vruusmann** [comment]
@pegger0709 I've released SkLearn2PMML version 0.49.3 to PyPI just now. It adds support for two more aggregation functions - `sum` and `product`.
